modules/slo/main.tf

```diff
@@ -17,32 +17,45 @@
 locals {
   full_name                 = "slo-${var.config.service_name}-${var.config.feature_name}-${var.config.slo_name}"
   pubsub_configs            = [for e in var.config.exporters : e if lower(e.class) == "pubsub"]
-  function_source_directory = var.function_source_directory != "" ? var.function_source_directory : "${path.module}/code"
   suffix                    = random_id.suffix.hex
+  function_source_directory = var.function_source_directory != "" ? var.function_source_directory : "${path.module}/code"
+  function_target_directory = "${path.module}/code_${local.suffix}"
   requirements_txt = templatefile(
     "${path.module}/code/requirements.txt.tpl", {
       slo_generator_version = var.slo_generator_version
     }
   )
 }
 
+resource "null_resource" "copy_gcf_folder" {
```

Comment:
I'm not sure this is necessary? Last I checked the `local_file` resource autocreates parent directories.

Reply:
ah, indeed it does, had forgotten that.